### PR TITLE
Add support for bulk indexing

### PR DESCRIPTION
**Pull request by mhl10**

### CODE REVIEW

File: search/services/index.py [hunk 1]
```diff
@@ -355,24 +359,21 @@ def _prepare_simple(self, query: SimpleQuery) -> Search:
...
+        q = Q()
+        for au in query.authors:
+            _q = _Q('match', 'authors__last_name__folded', au.surname)
+
+            if au.forename:    # Try as both forename and initials.
                 _q_init = Q()
-                for i in author.forename.replace('.', ' ').split():
-                    _q_init &= _Q('match', f'{field}__initials__folded', i)
+                for i in au.forename.replace('.', ' ').split():
+                    _q_init &= _Q('match', 'authors__initials__folded', i)
```
Comment (erickpeirson) [inline search/services/index.py]: Looks like this is reverting some recent work

File: search/services/index.py [hunk 2]
```diff
@@ -355,24 +359,21 @@ def _prepare_simple(self, query: SimpleQuery) -> Search:
...
                 _q &= (
-                    _Q('match', f'{field}__first_name__folded',
-                       author.forename)
-                    | _q_init
+                    _Q('match', 'authors__first_name__folded', au.forename)
```
Comment (erickpeirson) [inline search/services/index.py]: This too

File: bulk_index.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    IndexingFailed
+
+app = create_ui_web_app()
+app.app_context().push()
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
```
Comment (erickpeirson) [inline bulk_index.py]: You might have a look at https://docs.python.org/3/library/itertools.html#itertools.tee -- I am constantly (re)discovering goodies in itertools

File: search/services/index.py [hunk 4]
```diff
@@ -355,24 +359,21 @@ def _prepare_simple(self, query: SimpleQuery) -> Search:
         current_search = self._apply_sort(query, current_search)
         return current_search
 
-    def _author_query_part(self, author: Author, field: str) -> Search:
-        _q = None
-        if author.surname:
-            _q = _Q('match', f'{field}__last_name__folded', author.surname)
-            if author.forename:    # Try as both forename and initials.
+    def _prepare_author(self, query: AuthorQuery) -> Search:
```
Comment (erickpeirson) [inline search/services/index.py]: This looks reverted as well

File: bulk_index.py (hunk 3 above)
Comment (mhl10) [reply bulk_index.py]: Thanks, lots of goodies indeed! `itertools.tee` should work fine here.

**Comment by mhl10** [comment]
Mea culpa regarding search/services/index.py; I believe that's now fixed. I'll address the code-quality issues separately.

**Comment by erickpeirson** [comment]
No prob, looks good pending style fixes

**Comment by erickpeirson** [comment]
Fixes #72

**Pull request merged**